contribs/gnodev/pkg/dev/node.go

```diff
@@ -294,31 +293,37 @@ func (d *Node) GetBlockTransactions(blockNum uint64) ([]std.Tx, error) {
 // GetBlockTransactions returns the transactions contained
 // within the specified block, if any
 // GetLatestBlockNumber returns the latest block height from the chain
-func (d *Node) GetLatestBlockNumber() (uint64, error) {
-	return d.getLatestBlockNumber(), nil
+func (n *Node) GetLatestBlockNumber() (uint64, error) {
+	n.muNode.RLock()
+	defer n.muNode.RUnlock()
+
+	return n.getLatestBlockNumber(), nil
 }
 
 // SendTransaction executes a broadcast commit send
 // of the specified transaction to the chain
-func (d *Node) SendTransaction(tx *std.Tx) error {
+func (n *Node) SendTransaction(tx *std.Tx) error {
+	n.muNode.RLock()
+	defer n.muNode.RUnlock()
+
 	aminoTx, err := amino.Marshal(tx)
 	if err != nil {
 		return fmt.Errorf("unable to marshal transaction to amino binary, %w", err)
 	}
 
 	// we use BroadcastTxCommit to ensure to have one block with the given tx
-	res, err := d.client.BroadcastTxCommit(aminoTx)
+	res, err := n.client.BroadcastTxCommit(aminoTx)
```

Comment:
Just a note, if you ever want to make this async in the future, you can do a sync broadcast, and poll for the tx result on the `Tx` endpoint, using the tx hash

Reply:
I think for now I want to have this sync as much as possible to simplify the flow, but nice tips, thanks !